blockchain/blockchain.go

```diff
@@ -366,7 +366,7 @@ func verifyBlock(txn db.Transaction, block *core.Block) error {
 // "[]" is the db prefix to represent a bucket
 // "()" are additional keys appended to the prefix or multiple values marshalled together
 // "->" represents a key value pair.
-func storeBlockHeader(txn db.Transaction, header *core.Header) error {
+func StoreBlockHeader(txn db.Transaction, header *core.Header) error {
```

Comment:
I am still a bit reluctant to do that. I think a single package that handles transaction lifetime is important, since it is pretty error prone IMO.

Another alternative could be a separate package called something like `starknetdb`  that exposes functions that work with transactions and blockchain still handles the transaction lifetimes

Reply:
> a single package that handles transaction lifetime is important

Good point. I think we only need to be concerned about write transaction lifetime. Since the synchronizer is the only entity that needs write access (besides migrations), maybe some helper functions or even an `organizer` package can help manage the complexity?